### ISSUE TITLE
about the indent and background

### ISSUE THREAD

**Issue by phoeller** [EnlighterJS/Plugin.WordPress#203]
Hi there,
  I got EnglihterJS installed in the wordpress, and copy and paste the example for testing. I am using TinyMCE Advance text mode for my page and I have the following two pieces of scripts inputted

> < pre class="EnlighterJSRAW" data-enlighter-theme="atomic" data-enlighter-language="python">
> def Fib(n):
>     return 1 if n<=2 else Fib(n-1)+Fib(n-2)
> print(Fib(int(input())))
> </pre>
> < p>< code class="EnlighterJSRAW" data-enlighter-theme="atomic" data-enlighter-language="python">def Fib(n):<br />
>     return 1 if n<=2 else Fib(n-1)+Fib(n-2)
> print(Fib(int(input())))
> </code></p>

Here is the link of [example](http://icecues.com/index.php/2020/01/17/test/)

The first one looks good in style but with the wrong indent. If I change it to raw code mode or seen that in block editor mode, the indent looks good. 

The second one, the indent is correct but the background totally messed up. I saw one of the online demo to show to use < code> instead of < pre>, I thus change pre to code block but won't work.

**Comment by AndiDittrich** [comment]
Hi @phoeller,

`pre` tags are intended to be used for **blocks** and `code` tags for single line inline code (see enlighterjs.org)

your example above doesn't contain any indentation - can you please post a valid example ?

----

btw. blocking right click mouse-events is absolutely useless for advanced users...

**Comment by phoeller** [comment]
Hi AndiDittrich,
  In the example given in the [link](http://icecues.com/index.php/2020/01/17/test/), I copy and paste the code from Python editor. I add one more example of matlab by using TAB to insert an indent and paste the code to pre block. In the visual mode of TinyMCE, TAB key doesn't work. In matlab editor, I saw that the command disp with "d" align with the "n" in the for statement. But in the pre block, they are not lining up. However, if you switch the pre block to RAW code mode, alignment is correct.

**Comment by AndiDittrich** [comment]
i cannot see any differences between raw mode and regular highlighting. just the tab indent char is replaced with spaces (this can be controlled within the options).

the enlighter TinyMCE plugin includes a tab-indentation mode: it allows you to use <kbd>tab</kbd> as well as <kbd>shift+tab</kbd>

btw. it is recommend to use the "Code Insert Dialog" of Enlighter in case you're using the visual editor mode

**Comment by phoeller** [comment]
Thank Andi, I still don't know how it works. But I did several experiments and I find that if I copy-paste the code with indents by using Code Insert Dialog into the tinyMCE visual or text mode, all idents are gone so no matter what code-indent I choose, it won't show the correct indented code.

It is interesting that if I copy and paste the same code into [Classic Editor Live-Demo](https://tinymce.enlighterjs.org/) by using Code Insert Dialog, it always shows the correct indented result. I wonder if any difference between my TinyMCE and the one used in the  Classic Editor Live-Demo

As for the plugin allow tab and shift-tab, where could I find it? I did some search but I don't see where is it.
